Spanner/src/Operation.php

```diff
@@ -224,7 +230,13 @@ public function execute(Session $session, $sql, array $options = [])
 
         $context = $this->pluck('transactionContext', $options);
 
-        $call = function ($resumeToken = null) use ($session, $sql, $options) {
+        // Initially with begin, transactionId will be null.
+        // Once generated it will get populated from Result->rows()
+        // Incase of stream failure, the transactionId is preserved here to reuse.
+        $call = function ($resumeToken = null) use ($session, $sql, $options, &$transactionId) {
+            if ( !is_null($transactionId) and !is_array($transactionId) ) {
```

Comment:
You could have better type-checking by using `is_string`
```suggestion
            if (is_string($transactionId)) {
```